src/Helmet.js

```diff
@@ -232,17 +267,33 @@ const generateTagsAsReactComponent = (type, tags) => {
     /* eslint-enable react/display-name */
 };
 
-const getMethodsForTag = (type, tags) => ({
-    toComponent: (type === TAG_NAMES.TITLE) ? () => generateTitleAsReactComponent(type, tags) : () => generateTagsAsReactComponent(type, tags),
-    toString: (type === TAG_NAMES.TITLE) ? () => generateTitleAsString(type, tags) : () => generateTagsAsString(type, tags)
-});
+const getMethodsForTag = (type, tags) => {
+    switch (type) {
```

Comment:
To DRY it up and still keep it clear, what do you think about having title and default be the same case:

```
case TAG_NAMES.HTML:
return {...};

case TAG_NAMES.TITLE:
case default:
return {...};

```